plugins/woocommerce/includes/admin/class-wc-admin-meta-boxes.php

```diff
@@ -226,24 +226,31 @@ public function save_meta_boxes( $post_id, $post ) {
 	}
 
 	/**
-	 * Remove block-based templates from the list of available templates for products.
+	 * Remove irrelevant block templates from the list of available templates for products.
+	 * This will also remove custom created templates.
 	 *
 	 * @param string[] $templates Array of template header names keyed by the template file name.
 	 *
 	 * @return string[] Templates array excluding block-based templates.
 	 */
-	public function remove_block_templates( $templates ) {
-		if ( count( $templates ) === 0 || ! function_exists( 'gutenberg_get_block_template' ) ) {
+	public function filter_product_block_templates( $templates ) {
+		if ( count( $templates ) === 0 || ! wc_current_theme_is_fse_theme() || ! function_exists( 'gutenberg_get_block_template' ) ) {
 			return $templates;
 		}
 
 		$theme              = wp_get_theme()->get_stylesheet();
 		$filtered_templates = array();
 
 		foreach ( $templates as $template_key => $template_name ) {
-			$gutenberg_template = gutenberg_get_block_template( $theme . '//' . $template_key );
+			// Filter out the single-product.html template as this is a duplicate of "Default Template".
+			if ( 'single-product' === $template_key ) {
+				continue;
+			}
+
+			$block_template = gutenberg_get_block_template( $theme . '//' . $template_key );
 
-			if ( ! $gutenberg_template ) {
+			// If the block template has the product post type specified, include it.
+			if ( in_array( 'product', $block_template->post_types ) ) {
```

Comment:
Do we need to account for the case the template is not a block template (ie `gutenberg_get_block_template()` returns null)?

Reply:
Ah yes thanks for pointing this out.